### PR TITLE
Use plain socket objects instead of wrapper classes

### LINKED ISSUE #3013: Bad file descriptor problem still exists as of today
Hello,

why was this
https://github.com/benoitc/gunicorn/issues/1877
closed and further comments after closure completely ignored?

I must admit the crosslinking between the project issues and the conversation is extremely hard to follow, for me honestly not at all anymore now what is done, reverted, has to be done, etc.
https://github.com/encode/uvicorn/pull/1710

Can someone please shed some light onto if and how this will be fixed, because this is still an issue as of today with gunicorn 20.1.- and uvicorn / uvicornworkers 0.22, and is also very easy to reproduce every single time the server is shutdown (e.g. with Ctrl-C)

Maybe 
@Kludex ??

Answer highly appreciated, thanks

### PR DESCRIPTION
**Pull request by tilgovi**
Refactor socket creation to remove the socket wrapper classes so that these objects have less surprising behavior when used in worker hooks, worker classes, and custom applications.

Close #3013.

### CODE REVIEW

**Comment by jsprieto10** [comment]
is there anything else missing for this?

**Comment by tilgovi** [comment]
@benoitc polite bump, if you have the time

File: gunicorn/sock.py [hunk 1]
```diff
@@ -14,130 +14,56 @@
...
+            return sock
+        except socket.error as e:
+            if e.args[0] == errno.EADDRINUSE:
+                log.error("Connection in use: %s", str(addr))
+            if e.args[0] == errno.EADDRNOTAVAIL:
+                log.error("Invalid address: %s", str(addr))
+            if i < 5:
+                msg = "connection to {addr} failed: {error}"
+                log.debug(msg.format(addr=str(addr), error=str(e)))
+                log.error("Retrying in 1 second.")
```
[the remark this answers is not in the document]
Comment (benoitc) [reply gunicorn/sock.py:62]: @pajod what's your idea there?

**Comment by benoitc** [comment]
@tilgovi is this ready for review or do yui think to any other addition/changes already?

**Comment by tnusser** [comment]
@tilgovi What is the status of your PR?

File: gunicorn/sock.py [hunk 2]
```diff
@@ -150,67 +76,71 @@ def create_sockets(conf, log, fds=None):
...
+        listeners = []
+        for fd in list(fds) + [a for a in conf.address if isinstance(a, int)]:
+            sock = socket.socket(fileno=fd)
+            set_socket_options(conf, sock)
+            listeners.append(sock)
+    else:
+        # first initialization of gunicorn
+        old_umask = os.umask(conf.umask)
+        try:
+            for addr in [bind for bind in conf.address if not isinstance(bind, int)]:
```
Comment (benoitc) [inline gunicorn/sock.py:90]: can you make this part more readable. I would create the bind list  in a separate line there.

**Review by benoitc — CHANGES_REQUESTED** [review]
@tilgovi can you handle the change there. Otherwise I think it's OK for merge.

File: gunicorn/sock.py (hunk 2 above)
Comment (tnusser) [reply gunicorn/sock.py:90]: Does my [PR](https://github.com/benoitc/gunicorn/pull/3188) work for you?

**Comment by benoitc** [comment]
Closing this draft PR due to inactivity. Please feel free to reopen when you're ready to continue working on it. Thank you for your contribution\!
